app/models/operation.rb

```diff
@@ -1,17 +1,25 @@
+# This class only maintains the state of an operation and should
```

Comment:
This is a good start but can you add a sentence before this just introducing what an operation is?